chia/consensus/block_header_validation.py

```diff
@@ -491,6 +491,32 @@ def validate_unfinished_header_block(
     if q_str is None:
         return None, ValidationError(Err.INVALID_POSPACE)
 
+    # 5c. Check plot id is not present within last `NUM_DISTINCT_CONSECUTIVE_PLOT_IDS` blocks.
+    if height >= constants.SOFT_FORK3_HEIGHT:
+        curr_optional_block_record: Optional[BlockRecord] = prev_b
+        plot_id = get_plot_id(header_block.reward_chain_block.proof_of_space)
+        passed_previous_sp: bool = False
+        seen_signage_points: Set[bytes32] = {cc_sp_hash}
+
+        while (
+            curr_optional_block_record is not None
+            and len(seen_signage_points) < constants.NUM_PLOT_FILTERS_DISALLOWED_TO_PASS
+        ):
+            if passes_plot_filter(
+                constants,
+                plot_id,
+                curr_optional_block_record.pos_ss_cc_challenge_hash,
```

Comment:
iirc the attack we where talking about is inside of one slot so the challenge chain hash (pos_ss_cc_challenge_hash) should be the same, we dont have to save it inside BlockRecord